mycore-base/src/main/java/org/mycore/common/content/MCRSourceContent.java

```diff
@@ -117,6 +69,58 @@ public MCRSourceContent(Source source) {
         this.setBaseContent(baseContent);
     }
 
+    private static MCRContent getBaseContent(StreamSource streamSource) {
+        InputStream inputStream = streamSource.getInputStream();
+        if (inputStream != null) {
+            return new MCRStreamContent(inputStream);
+        } else {
+            try {
+                URI uri = new URI(streamSource.getSystemId());
+                return new MCRURLContent(uri.toURL());
+            } catch (URISyntaxException | MalformedURLException e) {
+                throw new MCRException("Could not create instance of MCRURLContent for SYSTEMID: "
+                    + streamSource.getSystemId(), e);
+            }
+        }
+    }
+
+    private static MCRByteContent getBaseContent(JAXBSource source) {
+        TransformerFactory transformerFactory = TransformerFactory.newDefaultInstance();
+        try {
+            Transformer transformer = transformerFactory.newTransformer();
+            ByteArrayOutputStream bout = new ByteArrayOutputStream();
+            transformer.transform(source, new StreamResult(bout));
+            return new MCRByteContent(bout.toByteArray());
+        } catch (TransformerException e) {
+            throw new MCRException("Error while resolving JAXBSource", e);
+        }
+    }
+
+    private static MCRJDOMContent getBaseContent(JDOMSource src) {
+        Document xml = src.getDocument();
+        if (xml != null) {
+            return new MCRJDOMContent(xml);
+        } else {
+            for (Object node : src.getNodes()) {
+                if (node instanceof Element element) {
+                    Document doc = element.getDocument();
+                    if (doc == null) {
+                        return new MCRJDOMContent(element);
+                    } else {
```

Comment:
Not required else case

Reply:
Sorry. Do not understand why. Can you explain please?

Reply:
When doc is equal null, it is returned, therefore the else is not required:
```
if (doc == null) {
  return new MCRJDOMContent(element);
} 
if (doc.getRootElement() == element) {
  return new MCRJDOMContent(doc);
} else {
  return new MCRJDOMContent(element.clone());
}
```
is sufficient, removes one nesting

Reply:
Should be addressed in https://github.com/MyCoRe-Org/mycore/pull/2050/commits/41defe31da69bae47ba323e2c9337f46ae90e3a6